scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorTask.java

```diff
@@ -500,8 +505,21 @@ public void insertJob(final ActorJob job) {
     fastLaneJobs.addFirst(job);
   }
 
-  public void fail() {
-    lifecyclePhase = ActorLifecyclePhase.FAILED;
+  public void fail(final Throwable error) {
+    final var previousPhase = LIFECYCLE_UPDATER.getAndSet(this, ActorLifecyclePhase.FAILED);
+
+    if (previousPhase == ActorLifecyclePhase.FAILED) {
+      return;
+    }
+
+    if (previousPhase == ActorLifecyclePhase.STARTING) {
+      startingFuture.completeExceptionally(error);
+    }
+
+    if (previousPhase != ActorLifecyclePhase.CLOSED) {
+      closeFuture.complete(null);
```

Comment:
:thought_balloon: Since closeFuture is completed in multiple places now, I was wondering if we ever try to complete already completed future. I guess not, but just want to think it aloud.